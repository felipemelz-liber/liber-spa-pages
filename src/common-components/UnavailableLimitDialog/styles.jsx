import styled from 'styled-components';
import { FlatButton } from 'liber-components';

const TEXT_COLOR = '#6287a7';

export const Container = styled.div`
  padding: 24px 24px 12px 24px;
  width: 480px;

  * {
    font-family: Roboto;
  }
`;

export const Header = styled.div`
  font-size: 21px;
  font-weight: bold;
  line-height: 1.4;
  letter-spacing: 0.08px;
  color: ${TEXT_COLOR};
  margin-bottom: 24px;
`;

export const Content = styled.div`
  font-size: 16px;
  line-height: 1.38;
  letter-spacing: 0.01px;
  margin-bottom: 24px;
  color: ${TEXT_COLOR};
  margin-bottom: 24px;
`;

export const ActionsRow = styled.div`
  display: flex;
  justify-content: flex-end;
`;

export const Button = styled(FlatButton).attrs(() => ({ size: 'large' }))``;
