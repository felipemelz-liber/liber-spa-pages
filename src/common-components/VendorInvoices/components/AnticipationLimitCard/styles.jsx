import styled from 'styled-components';

export const Card = styled.div`
  font-family: Roboto;
  color: #6287a7;
  width: 360px;
  box-shadow: 0px 3px 6px #00000029;
  border: 1px solid #cdd7df;
  border-radius: 8px;
  padding: 16px 24px;
  margin-bottom: 56px;

  * {
    font-family: Roboto;
  }
`;

export const CardTitle = styled.div`
  font-size: 16px;
  line-height: 22px;
  letter-spacing: 0.06px;
  font-weight: 500;
  color: #4b6f85;
  margin-bottom: 4px;
`;
