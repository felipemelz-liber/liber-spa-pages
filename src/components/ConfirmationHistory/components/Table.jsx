import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { CompleteTable } from 'liber-components';
import { DropDownContent, DropDownContentRow, EyeIcon, Global } from './Table.styles.jsx';
import { CONFIRMATION_HIST_COLUMNS, filtersPropTypes } from '../utils';
import { updatePagination } from '../actions/index.js';
import Filters from './filters/index.js';

export const ConfirmationHistoryTable = ({
  confirmations,
  handleFilter,
  pagination,
  onUpdatePagination,
  token,
  filters,
}) => {
  const { current, per, pages, count } = pagination;

  const listingStart = () => String((current - 1) * per + 1);
  const listingEnd = () => {
    const end = (current - 1) * per + per;
    if (count > end) return String(end);
    return String(count);
  };

  return (
    <>
      <Global />
      <CompleteTable
        columns={CONFIRMATION_HIST_COLUMNS}
        items={confirmations}
        filters={<Filters count={count} handleFilter={handleFilter} />}
        actions={({ details }) => (
          <DropDownContent>
            <DropDownContentRow href={details}>
              <EyeIcon />
              Visualizar Confirmação
            </DropDownContentRow>
          </DropDownContent>
        )}
        paginationProps={{
          currentPage: String(current),
          pageTotal: String(pages),
          onChangePage: target =>
            onUpdatePagination(token, { per, page: parseInt(target.value, 10), ...filters }),
          onClickNext: () => onUpdatePagination(token, { per, page: current + 1, ...filters }),
          onClickPrev: () => onUpdatePagination(token, { per, page: current - 1, ...filters }),
          hasListing: true,
          listingProps: {
            onSelectListing: selectedListing =>
              onUpdatePagination(token, {
                per: parseInt(selectedListing, 10),
                page: 1,
                ...filters,
              }),
            listing: String(per),
            total: String(count),
            listingLabelSingular: 'confirmação',
            listingLabelPlural: 'confirmações',
            showInterval: {
              start: listingStart(),
              end: listingEnd(),
            },
          },
          onKeyDownPage: () => null,
        }}
      />
    </>
  );
};

ConfirmationHistoryTable.propTypes = {
  confirmations: PropTypes.arrayOf(PropTypes.object),
  handleFilter: PropTypes.func,
  pagination: PropTypes.shape({
    current: PropTypes.number,
    per: PropTypes.number,
    pages: PropTypes.number,
    count: PropTypes.number,
  }),
  onUpdatePagination: PropTypes.func,
  token: PropTypes.string.isRequired,
  filters: filtersPropTypes.isRequired,
};

ConfirmationHistoryTable.defaultProps = {
  confirmations: [],
  handleFilter: () => null,
  pagination: {},
  onUpdatePagination: () => null,
};

const mapStateToProps = ({ confirmationHistory }) => ({
  confirmations: confirmationHistory.confirmations,
  pagination: confirmationHistory.pagination,
  token: confirmationHistory.token,
  filters: confirmationHistory.filters,
});

const mapDispatchToProps = {
  onUpdatePagination: updatePagination,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(ConfirmationHistoryTable);
