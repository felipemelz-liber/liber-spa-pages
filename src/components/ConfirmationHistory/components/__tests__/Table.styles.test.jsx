import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';
import * as styles from '../Table.styles';

describe('Testing ConfirmationHistoryTable styles', () => {
  configure({ adapter: new Adapter() });
  const { TableRowColumn, ...uncontrolled } = styles;
  Object.keys(uncontrolled).forEach(style => {
    it(`renders ${style} correctly`, () => {
      const App = uncontrolled[style];
      const wrapper = shallow(<App />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
  it('renders TableRowColumn correctly without props', () => {
    const wrapper = shallow(<TableRowColumn />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('renders TableRowColumn correctly with right aligment', () => {
    const wrapper = shallow(<TableRowColumn align="right" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('renders TableRowColumn correctly with center aligment', () => {
    const wrapper = shallow(<TableRowColumn align="center" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('renders TableRowColumn correctly with left aligment', () => {
    const wrapper = shallow(<TableRowColumn align="left" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
