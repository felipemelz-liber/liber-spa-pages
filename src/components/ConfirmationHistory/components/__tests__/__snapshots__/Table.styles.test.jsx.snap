// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ConfirmationHistoryTable styles renders ActionHeaderContainer correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-EHOje",
        "isStatic": false,
        "rules": Array [
          "
  width: 100%;
  display: flex;
  justify-content: center;
  margin-right: 17px;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-EHOje",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Testing ConfirmationHistoryTable styles renders DownloadIcon correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-gqjmRU",
        "isStatic": false,
        "rules": Array [
          "
  fill: #009dff;
",
        ],
      },
      "displayName": "Styled(MdDownload)",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-gqjmRU",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Testing ConfirmationHistoryTable styles renders DropDownButton correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bwzfXH",
        "isStatic": false,
        "rules": Array [
          "
  width: 32px;
  height: 32px;
  padding: 8px;
  background-color: #ecf0f3;
",
        ],
      },
      "displayName": "Styled(Buttons__Button)",
      "foldedComponentIds": Array [
        "Buttons__Button-u35aut-0",
      ],
      "render": [Function],
      "styledComponentId": "sc-bwzfXH",
      "target": Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      },
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Testing ConfirmationHistoryTable styles renders DropDownContent correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-htpNat",
        "isStatic": false,
        "rules": Array [
          "
  display: flex;
  flex-direction: column;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-htpNat",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Testing ConfirmationHistoryTable styles renders DropDownContentRelative correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-ifAKCX",
        "isStatic": false,
        "rules": Array [
          "
  position: relative;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-ifAKCX",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Testing ConfirmationHistoryTable styles renders DropDownContentRow correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bxivhb",
        "isStatic": false,
        "rules": Array [
          "
  display: flex;
  flex-direction: row;
  padding: 4px;
  color: inherit;
  & > svg {
    margin-right: 8px;
    margin-top: 4px;
  }
  :hover {
    color: #009dff;
    & > svg {
      fill: #009dff;
    }
  }
",
        ],
      },
      "displayName": "styled.a",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bxivhb",
      "target": "a",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Testing ConfirmationHistoryTable styles renders EyeIcon correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bZQynM",
        "isStatic": false,
        "rules": Array [
          "
  width: 16px;
  height: 16px;
  fill: #4b6f85;
",
        ],
      },
      "displayName": "Styled(MdEye)",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bZQynM",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Testing ConfirmationHistoryTable styles renders Global correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`Testing ConfirmationHistoryTable styles renders MoreVertButtonContainer correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-iwsKbI",
        "isStatic": false,
        "rules": Array [
          "
  position: relative;
  padding: 4px 0px;
  width: 32px;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-iwsKbI",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Testing ConfirmationHistoryTable styles renders MoreVertIcon correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  width: 16px;
  height: 16px;
",
        ],
      },
      "displayName": "Styled(MoreVert)",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Testing ConfirmationHistoryTable styles renders StyledButton correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-gZMcBi",
        "isStatic": false,
        "rules": Array [
          "
  font-family: Roboto;
  font-size: 16px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: 1.38;
  letter-spacing: 0.06px;
  color: #009dff;
  background-color: transparent;
  margin-left: -18px;
  & > svg {
    fill: #009dff;
    margin-right: 4px;
  }
  :hover {
    color: #6287a7;
    & > svg {
      fill: #6287a7;
    }
  }
",
        ],
      },
      "displayName": "Styled(Buttons__Link)",
      "foldedComponentIds": Array [
        "Buttons__Link-u35aut-6",
      ],
      "render": [Function],
      "styledComponentId": "sc-gZMcBi",
      "target": Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      },
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Testing ConfirmationHistoryTable styles renders Table correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-gzVnrw",
        "isStatic": false,
        "rules": Array [
          "min-width:100%;border-collapse:collapse;table-layout:fixed;font-family:Roboto;text-align:left;font-size:20px;line-height:24px;position:relative;",
          "
  width: 100%;
",
        ],
      },
      "displayName": "Styled(SelectionTable)",
      "foldedComponentIds": Array [
        "SelectionTable-x9uakv-0",
      ],
      "render": [Function],
      "styledComponentId": "sc-gzVnrw",
      "target": "table",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Testing ConfirmationHistoryTable styles renders TableRow correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-htoDjs",
        "isStatic": false,
        "rules": Array [
          "--row-color:",
          [Function],
          ";border:0px;position:relative;",
          ".SelectionTable__SelectionTableBody-x9uakv-2",
          " &{border-top:1px solid #ecf0f3;",
          [Function],
          " &:hover{background-color:#f3f3f3;}}",
          "
  display: flex;
  width: 100%;
  justify-content: space-between;
  &:first-of-type {
    border-top: none;
  }
",
        ],
      },
      "displayName": "Styled(SelectionTable__SelectionTableRow)",
      "foldedComponentIds": Array [
        "SelectionTable__SelectionTableRow-x9uakv-5",
      ],
      "render": [Function],
      "styledComponentId": "sc-htoDjs",
      "target": "tr",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Testing ConfirmationHistoryTable styles renders TableRowColumn correctly with center aligment 1`] = `
<StyledComponent
  align="center"
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-dnqmqq",
        "isStatic": false,
        "rules": Array [
          "min-height:48px;border:0px;padding:8px;color:var(--row-color);position:relative;font-size:16px;",
          "
  display: flex;
  height: 100%;
  align-items: center;
  justify-content: flex-end;
  ",
          [Function],
          "
",
        ],
      },
      "displayName": "Styled(SelectionTable__SelectionTableRowColumn)",
      "foldedComponentIds": Array [
        "SelectionTable__SelectionTableRowColumn-x9uakv-4",
      ],
      "render": [Function],
      "styledComponentId": "sc-dnqmqq",
      "target": "td",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Testing ConfirmationHistoryTable styles renders TableRowColumn correctly with left aligment 1`] = `
<StyledComponent
  align="left"
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-dnqmqq",
        "isStatic": false,
        "rules": Array [
          "min-height:48px;border:0px;padding:8px;color:var(--row-color);position:relative;font-size:16px;",
          "
  display: flex;
  height: 100%;
  align-items: center;
  justify-content: flex-end;
  ",
          [Function],
          "
",
        ],
      },
      "displayName": "Styled(SelectionTable__SelectionTableRowColumn)",
      "foldedComponentIds": Array [
        "SelectionTable__SelectionTableRowColumn-x9uakv-4",
      ],
      "render": [Function],
      "styledComponentId": "sc-dnqmqq",
      "target": "td",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Testing ConfirmationHistoryTable styles renders TableRowColumn correctly with right aligment 1`] = `
<StyledComponent
  align="right"
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-dnqmqq",
        "isStatic": false,
        "rules": Array [
          "min-height:48px;border:0px;padding:8px;color:var(--row-color);position:relative;font-size:16px;",
          "
  display: flex;
  height: 100%;
  align-items: center;
  justify-content: flex-end;
  ",
          [Function],
          "
",
        ],
      },
      "displayName": "Styled(SelectionTable__SelectionTableRowColumn)",
      "foldedComponentIds": Array [
        "SelectionTable__SelectionTableRowColumn-x9uakv-4",
      ],
      "render": [Function],
      "styledComponentId": "sc-dnqmqq",
      "target": "td",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Testing ConfirmationHistoryTable styles renders TableRowColumn correctly without props 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-dnqmqq",
        "isStatic": false,
        "rules": Array [
          "min-height:48px;border:0px;padding:8px;color:var(--row-color);position:relative;font-size:16px;",
          "
  display: flex;
  height: 100%;
  align-items: center;
  justify-content: flex-end;
  ",
          [Function],
          "
",
        ],
      },
      "displayName": "Styled(SelectionTable__SelectionTableRowColumn)",
      "foldedComponentIds": Array [
        "SelectionTable__SelectionTableRowColumn-x9uakv-4",
      ],
      "render": [Function],
      "styledComponentId": "sc-dnqmqq",
      "target": "td",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
